### ISSUE TITLE
Make PhraseQuery faster

### ISSUE THREAD

**Issue by fulmicoton** [quickwit-oss/tantivy#125]
Phrase query are currently very slow.
The current implementation require to uncompress the full positions list upfront.

You might want to check what is done in Lucene and in the literature. (skip lists? index ngrams?)
Let's document in this ticket the options, with links and original source.

**Comment by fulmicoton** [comment]
Depends on #11

**Comment by fulmicoton** [comment]
Duplicate of #248
